### PR TITLE
changed threading handling

### PR DESCRIPTION
**Pull request by JensWehner**
Changed the default handling

### CODE REVIEW

File: CMakeModules/FindMKL.cmake [hunk 1]
```diff
@@ -278,3 +283,5 @@ if(MKL_FOUND)
         IMPORTED_IMPLIB "${MKL_Core_LINK_LIBRARY}")
   endif()
 endif()
+
+message("-- " ${MKL_THREAD_LAYER} " used for MKL parallelisation")
```
Comment (junghans) [inline CMakeModules/FindMKL.cmake]:
```suggestion
message(STATUS "'${MKL_THREAD_LAYER}' used for MKL parallelisation")
```

File: CMakeModules/FindMKL.cmake [hunk 2]
```diff
@@ -62,7 +62,12 @@ include(CheckTypeSize)
 check_type_size("int" INT_SIZE
   BUILTIN_TYPES_ONLY LANGUAGE C)
 
-set(MKL_THREAD_LAYER "Sequential" CACHE STRING "The thread layer to choose for MKL")
+
+if(NOT MKL_THREAD_LAYER)
+    set(MKL_THREAD_LAYER "TBB")
+endif()
+
+set(MKL_THREAD_LAYER ${MKL_THREAD_LAYER} CACHE STRING "The thread layer to choose for MKL")
```
Comment (junghans) [inline CMakeModules/FindMKL.cmake]:
```suggestion
set(MKL_THREAD_LAYER "TBB" CACHE STRING "The thread layer to choose for MKL")
```
User's choice will always overwrite.

File: CMakeModules/FindMKL.cmake [hunk 3]
```diff
@@ -62,7 +62,12 @@ include(CheckTypeSize)
 check_type_size("int" INT_SIZE
   BUILTIN_TYPES_ONLY LANGUAGE C)
 
-set(MKL_THREAD_LAYER "Sequential" CACHE STRING "The thread layer to choose for MKL")
+
+if(NOT MKL_THREAD_LAYER)
+    set(MKL_THREAD_LAYER "TBB")
+endif()
```
Comment (junghans) [inline CMakeModules/FindMKL.cmake]: Drop these 3 lines.

File: CMakeModules/FindMKL.cmake (hunk 2 above)
Comment (JensWehner) [reply CMakeModules/FindMKL.cmake]: Why did it not work before then?

Comment (JensWehner) [reply CMakeModules/FindMKL.cmake]: That line was exactly my problem.

Comment (junghans) [reply CMakeModules/FindMKL.cmake]: Before the default was `Sequential`, right?

Comment (JensWehner) [reply CMakeModules/FindMKL.cmake]: Yes but in tools CMakeList the parameter was explicitly set

Comment (JensWehner) [reply CMakeModules/FindMKL.cmake]: But FindMKL did not use it

Comment (junghans) [reply CMakeModules/FindMKL.cmake]: I see, `set(MKL_THREAD_LAYER "Intel OpenMP")` isn't enough, you need to do something like `set(MKL_THREAD_LAYER "Intel OpenMP" CACHE STRING ...)` there.

**Pull request merged**